From sathya.perla@emulex.com  Thu Jun 23 15:40:26 2011
From: Sathya Perla <sathya.perla@emulex.com>
Date: Thu, 12 May 2011 19:32:16 +0000
Subject: [PATCH] be2net: fix mbox polling for signal reception
Git-commit: 1dbf53a28262aa89ecbe653e8a9127c0baef9bc4
Patch-mainline: v3.0-rc1
References: FATE#311448, bnc#697255

Sending mbox cmds require multiple steps of writing to the DB register and polling
for an ack. Gettting interrupted in the middle by a signal breaks the mbox protocol.
Use msleep() to not get interrupted.

Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/benet/be_cmds.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)
--- a/drivers/net/benet/be_cmds.c
+++ b/drivers/net/benet/be_cmds.c
@@ -298,8 +298,7 @@ static int be_mbox_db_ready_wait(struct
 			return -1;
 		}
 
-		set_current_state(TASK_INTERRUPTIBLE);
-		schedule_timeout(msecs_to_jiffies(1));
+		msleep(1);
 		msecs++;
 	} while (true);
 
